### PR TITLE
Admin Menu: Set same menu order in both admin interfaces

### PR DESCRIPTION
**Pull request by mmtr**
Fixes DOTCOM-13889

## Proposed changes:

Ensures that all menu items are sorted in the same order for both the default/Calypso and the classic/wp-admin admin interfaces:
- Stops updating the position of the Jetpack menu in the default/Calypso admin interface, so it's kept in its default position (at the top)
- Changes the position of the Stats and Upgrades menu  in the default/Calypso admin interface, so they are placed before the Jetpack menu
- Stops hiding the default separator before Posts in the default/Calypso admin interface

Classic/wp-admin | Default/Calypso (Before) | Default/Calypso (After)
--- | --- | ---
Screenshot 2025-08-11 at 13 21 35 | Screenshot 2025-08-11 at 13 26 19 | Screenshot 2025-08-11 at 13 23 09

- Reorder the Tools submenus so they are sorted consistently across all interfaces:

Before | After
--- | ---
Screenshot 2025-08-11 at 17 38 45 | Screenshot 2025-08-11 at 17 23 57

- Reorder the Settings submenus so they are sorted consistently across all interfaces:

Before | After
--- | ---
Screenshot 2025-08-11 at 17 38 39 | Screenshot 2025-08-11 at 17 38 22

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
- Apply these changes to your WP.com site
- Activate the default admin interface
- Check the sidebar
- Make sure the Jetpack menu is placed between Upgrades and Posts
- Make sure there is a separator between Jetpack and Posts
- Make sure there are no regressions (e.g. menus duplicated or menus missing) when testing some edge scenarios:
  - Non-admins
  - P2 site
- Make sure there are no regressions in the classic admin interface
- Check that the submenu items of the Tools menu are sorted consistently across sites and admin interfaces
- Check that the submenu items of the Settings menu are sorted consistently across sites and admin interfaces

### CODE REVIEW

**Review by BogdanUngureanu — DISMISSED** [review]
Tests well! 🚢

**Comment by BogdanUngureanu** [comment]
Screenshot 2025-08-12 at 19 27 57

On Simple Sites the Monetize and Marketing are inversed for me :) 

It might be because of "Advertising". :)

**Comment by mmtr** [comment]
Good catch @BogdanUngureanu. It also happens on Atomic sites:

Simple / Default | Atomic / Default
--- | ---
Screenshot 2025-08-13 at 09 28 48 | Screenshot 2025-08-13 at 09 26 03

I don't know why it doesn't respect the order specified in the code (Marketing before Monetize), maybe I need to change the priority of `wpcom_add_tools_menu` 🤔

**Comment by mmtr** [comment]
> maybe I need to change the priority of `wpcom_add_tools_menu` 🤔

That was it. [Tools > Advertising was being registered](https://github.com/Automattic/jetpack/blob/c147cd4b63c1de30c44cf11f28a3ec50a7cd998e/projects/packages/blaze/src/class-blaze.php#L57) after we re-ordered the menus, so I had to delay the hook

**Comment by BogdanUngureanu** [comment]
That fixed the problem for that submenu. However, now `Import` is the last item in the list on Simple Sites. :) I haven't checked AT. 

 
Screenshot 2025-08-13 at 12 48 32

**Comment by mmtr** [comment]
> However, now `Import` is the last item in the list on Simple Sites. :) I haven't checked AT.

Another excellent catch! Fixed it in the last commit.

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-menu/wpcom-admin-menu.php [hunk 1]
```diff
@@ -638,7 +638,7 @@ function wpcom_add_tools_menu() {
 			'advertising',
 			'marketing',
 			'monetize',
-			'import.php',
+			'import',
```
Comment (BogdanUngureanu) [inline projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-menu/wpcom-admin-menu.php:641]: Does it work with the Classic interface? We use `import.php` there :D

Comment (mmtr) [reply projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-menu/wpcom-admin-menu.php:641]: Yes, it does! We match [any string that contains these slugs](https://github.com/Automattic/jetpack/blob/ce42a569c0f9da2a74a7098d76abc2d182cad25d/projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-menu/wpcom-admin-menu.php#L213), it's not an exact match.

**Review by BogdanUngureanu — APPROVED** [review]
LGTM!
